okhttp/src/main/java/okhttp3/internal/platform/Platform.java

```diff
@@ -187,8 +188,12 @@ public CertificateChainCleaner buildCertificateChainCleaner(SSLSocketFactory ssl
 
   public static boolean isConscryptPreferred() {
     // mainly to allow tests to run cleanly
-    if ("conscrypt".equals(System.getProperty("okhttp.platform"))) {
-      return true;
+    try {
+      if ("conscrypt".equals(System.getProperty("okhttp.platform"))) {
```

Comment:
should line 270 change also? anywhere we check any properties.

Reply:
OK, I've went through the code and updated other calls to System Properties as well.

Reply:
OK, I've went through the code and updated other calls to System Properties as well.

Reply:
What do you think about moving this pattern, attempt and swallow ACE to okhttp.internal.Util.getSystemProperty(String key, @Nullable String defaultValue).

Reply:
I agree, I've refactored the code and that made the pull request much cleaner.